examples/reaction/gray_scott_io.py

```diff
@@ -0,0 +1,89 @@
+"""
+Gray-Scott diffusion-reaction demo, taken from
+[Hundsdorf & Verwer 2003]. It also appears as a
+PETSc TS tutorial and can be found in your Firedrake
+installation at
+
+  $VIRTUAL_ENV/src/petsc/src/ts/tutorials/advection-diffusion-reaction/ex5.c
+
+The test case consists of two tracer species, which
+experience different diffusivities and react with one
+another. The problem is interesting from a numerical
+modelling perspective because of the nonlinear coupling
+between the two equations in the system. Instead of
+solving the equations as a monolithic system, we opt to
+solve them alternately using a Picard iteration.
+
+[Hundsdorf & Vermer 2003] W. Hundsdorf & J.G. Verwer
+    (2003). "Numerical Solution of Time-Dependent
+    Advection-Diffusion-Reaction Equations", Springer.
```

Comment:
Probably best to refer to the original demo version for this and just say how this particular script differs from it.

Reply:
Updated.